src/NuGet.Clients/PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -80,56 +95,59 @@ public static class ProjectJsonToPackageRefMigrator
 
         private static void MigrateRuntimes(PackageSpec packageSpec, Microsoft.Build.Evaluation.Project buildProject)
         {
+            ThreadHelper.ThrowIfNotOnUIThread();
             var runtimes = packageSpec.RuntimeGraph.Runtimes;
             var supports = packageSpec.RuntimeGraph.Supports;
             var runtimeIdentifiers = new List<string>();
             var runtimeSupports = new List<string>();
             if (runtimes != null && runtimes.Count > 0)
             {
                 runtimeIdentifiers.AddRange(runtimes.Keys);
-                
+
             }
 
-            if(supports != null && supports.Count > 0)
+            if (supports != null && supports.Count > 0)
             {
                 runtimeSupports.AddRange(supports.Keys);
             }
 
             var union = string.Join(";", runtimeIdentifiers.Union(runtimeSupports));
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
-            {
-                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                buildProject.SetProperty("RuntimeIdentifiers", union);
-            });            
+            buildProject.SetProperty("RuntimeIdentifiers", union);
         }
 
         private static void RemoveProjectJsonReference(Microsoft.Build.Evaluation.Project buildProject, string projectJsonFilePath)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
-            {
-                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                var projectJsonItem = buildProject.GetItems("None")
-                .Where(t => string.Equals(t.EvaluatedInclude, Path.GetFileName(projectJsonFilePath), StringComparison.OrdinalIgnoreCase))
-                .FirstOrDefault();
+            ThreadHelper.ThrowIfNotOnUIThread();
+            var projectJsonItem = buildProject.GetItems("None")
+            .Where(t => string.Equals(t.EvaluatedInclude, Path.GetFileName(projectJsonFilePath), StringComparison.OrdinalIgnoreCase))
+            .FirstOrDefault();
 
-                if (projectJsonItem != null)
-                {
-                    buildProject.RemoveItem(projectJsonItem);
-                }
-            });
+            if (projectJsonItem != null)
+            {
+                buildProject.RemoveItem(projectJsonItem);
+            }
         }
+        
 
-        private static void CreateBackup(LegacyCSProjPackageReferenceProject project,
+        private static async Task CreateBackupAsync(LegacyCSProjPackageReferenceProject project,
             string projectJsonFilePath)
         {
-            var backupDirectory = Path.Combine(Path.GetDirectoryName(project.MSBuildProjectPath), "Backup");
+            try
+            {
+                await TaskScheduler.Default;
+                var backupDirectory = Path.Combine(Path.GetDirectoryName(project.MSBuildProjectPath), "Backup");
```

Comment:
>Backup [](start = 103, length = 6)

my guess is that this should be localized. don't block on this now...but we should look at VS upgrader for guidance.